src/Test/BlockServiceTestCase.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\BlockBundle\Test;
+
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Sonata\BlockBundle\Block\BlockContextInterface;
+use Sonata\BlockBundle\Block\BlockContextManager;
+use Sonata\BlockBundle\Block\BlockContextManagerInterface;
+use Sonata\BlockBundle\Block\BlockLoaderInterface;
+use Sonata\BlockBundle\Block\BlockServiceInterface;
+use Sonata\BlockBundle\Block\BlockServiceManagerInterface;
+use Sonata\BlockBundle\Model\BlockInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Abstract test class for block service tests.
+ *
+ * @author Sullivan Senechal <soullivaneuh@gmail.com>
+ */
+abstract class BlockServiceTestCase extends TestCase
+{
+    /**
+     * @var MockObject|ContainerInterface
+     */
+    protected $container;
+
+    /**
+     * @var MockObject|BlockServiceManagerInterface
+     */
+    protected $blockServiceManager;
+
+    /**
+     * @var BlockContextManagerInterface
+     */
+    protected $blockContextManager;
+
+    /**
+     * @var FakeTemplating
+     */
+    protected $templating;
+
+    protected function setUp()
+    {
+        $this->container = $this->createMock(ContainerInterface::class);
+        $this->templating = new FakeTemplating();
+
+        $blockLoader = $this->createMock(BlockLoaderInterface::class);
+        $this->blockServiceManager = $this->createMock(BlockServiceManagerInterface::class);
+        $this->blockContextManager = new BlockContextManager($blockLoader, $this->blockServiceManager);
+    }
+
+    /**
+     * Create a mocked block service.
+     *
+     * @param BlockServiceInterface $blockService A block service
+     */
+    protected function getBlockContext(BlockServiceInterface $blockService): BlockContextInterface
```

Comment:
Should we use `final` here and below?

Reply:
We can't do this on the stable branch, also this are just test extension points :/

Reply:
It's not a BC-break: it's a new class, people are supposed to change their code to migrate anyway. But if those are extension points, then ok.